src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -120,6 +120,23 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}));
 	}
 
+	@HostListener('document:keydown', ['$event'])
+	onkeydown(e: KeyboardEvent) {
+		// Check that there are no active elements to prevent the execution of undo/redo commands in other elements.
+		// We need to make sure that there are no active cells to prevent triggering undo/redo within code cells and check that text cells are not in edit mode.
+		if (!this.areElementsActive() && (this.activeCellId === '' || !this.cellToolbar.first.getEditCellAction().editMode)) {
+			if ((e.ctrlKey || e.metaKey) && e.shiftKey && e.key === 'z') {
```

Comment:
ctrl+y is another commonly used redo shortcut (at least on Windows)

Reply:
https://github.com/microsoft/azuredatastudio/blob/7e17bfb6ec543a2141112bcdc681ce49134ab2ea/src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts#L78

Let's be consistent with text cell components with key bindings.

(Can we store these conditions in a common location to be accessed both by notebook component and text cell component?)

Reply:
@chlafreniere Should we use notebookCommon.ts for that? 

https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/common/notebookCommon.ts